### PR TITLE
Implement Boxplot aggregation

### PR DESCRIPTION
**Pull request by iamazy**

### CODE REVIEW

File: src/search/aggregations/metrics/boxplot_aggregation.rs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    #[serde(skip_serializing_if = "ShouldSkip::should_skip")]
+    compression: Option<Numeric>,
+    #[serde(skip_serializing_if = "ShouldSkip::should_skip")]
+    missing: Option<Numeric>,
+}
+
+impl Aggregation {
+    /// Creates an instance of [`BoxplotAggregation`]
+    ///
+    /// - `name` - name of the aggregation
```
Comment (buinauskas) [inline src/search/aggregations/metrics/boxplot_aggregation.rs:37]: Consider adding `/// - `field` - field to aggregate` as a doc. Doesn't add much value, but helps a little bit.

File: src/search/aggregations/metrics/boxplot_aggregation.rs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    /// 64KB in size. In practice data tends to be more random and the TDigest will use less memory.
+    pub fn compression(mut self, compression: impl Into<Numeric>) -> Self {
+        self.boxplot.compression = Some(compression.into());
+        self
+    }
+
+    /// The `missing` parameter defines how documents that are missing a value should be treated.
+    /// By default they will be ignored but it is also possible to treat them as if they had a value.
+    pub fn missing(mut self, missing: impl Into<Numeric>) -> Self {
+        self.boxplot.missing = Some(missing.into());
```
Comment (buinauskas) [inline src/search/aggregations/metrics/boxplot_aggregation.rs:73]: Can compression be a float or a negative value? Perhaps unsigned integer is a better candidate here?

Comment (iamazy) [reply src/search/aggregations/metrics/boxplot_aggregation.rs:74]: I checked the source code of es, the type of compression is `double`.

 https://github.com/elastic/elasticsearch/blob/20c9f756d28d116f0667e19a1b531aa80c55bde7/x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/boxplot/BoxplotAggregationBuilder.java#L53

maybe it's type should be `f64` here?

Comment (buinauskas) [reply src/search/aggregations/metrics/boxplot_aggregation.rs:74]: OK, I am convinced, thank you!

**Review by buinauskas — APPROVED** [review]
👏🏻

**Pull request merged**